common/build/eslint-plugin-fluid/src/rules/no-hyphen-after-jsdoc-tag.js

```diff
@@ -0,0 +1,61 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+/**
+ * JSDoc/TSDoc tags do not require a hyphen after them.
+ */
+module.exports = {
+	meta: {
+		type: "problem",
+		docs: {
+			description: "Disallow hyphen character immediately following JSDoc tag",
+			category: "Best Practices",
+			recommended: false,
+		},
+		messages: {
+			hyphenAfterTag:
+				"JSDoc/TSDoc block tags should not be followed by a hyphen character ('-').",
+		},
+		fixable: "code",
+		schema: [],
+	},
+
+	create(context) {
+		return {
+			Program() {
+				const sourceCode = context.getSourceCode();
+				const comments = sourceCode
+					.getAllComments()
+					// Filter to only JSDoc/TSDoc style block comments
+					.filter((comment) => comment.type === "Block" && comment.value.startsWith("*"));
+
+				for (const comment of comments) {
+					// Find any JSDoc/TSDoc tags followed by a hyphen
+					const matches = comment.value.matchAll(/(@[a-zA-Z0-9]+)\s+-(.*)/g);
```

Comment:
Isn't this causing `@param -` to also be reported as an error? Those can also show up in the same TSDoc block as the other tags, right?

Reply:
No, this should only catch cases where the hyphen immediately follows a tag description. Since `@param` tags are followed by a param name before the `-`, this should not trigger on those. I should add param comments to the test cases though. I'll do that.